tensorrt_llm/_torch/speculative/suffix.py

```
@@ -0,0 +1,208 @@
+        self.spec_config = spec_config
+        self.max_num_requests = max_num_requests
+        self.max_tree_depth = spec_config.max_tree_depth
+
+        # Lazy import to avoid error when Suffix Decoding is not used.
+        try:
+            from arctic_inference.suffix_decoding import SuffixDecodingCache
+        except ImportError:
+            raise ImportError(
+                "arctic_inference package is required for Suffix Decoding. "
+                "Please install it with: pip install arctic-inference"
+            )
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)